packages/components/src/modal/use-modal-exit-animation.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * WordPress dependencies
+ */
+import { useReducedMotion } from '@wordpress/compose';
+import { useCallback, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CONFIG } from '../utils';
+
+// Animation duration (ms) extracted to JS in order to be used on a setTimeout.
+const FRAME_ANIMATION_DURATION = CONFIG.transitionDuration;
+const FRAME_ANIMATION_DURATION_NUMBER = Number.parseInt(
+	CONFIG.transitionDuration
+);
+
+const EXIT_ANIMATION_NAME = 'components-modal__disappear-animation';
+
+export function useModalExitAnimation() {
+	const frameRef = useRef< HTMLDivElement >();
+	const [ isAnimatingOut, setIsAnimatingOut ] = useState( false );
+	const isReducedMotion = useReducedMotion();
+
+	const closeModal = useCallback(
+		() =>
+			new Promise< void >( ( closeModalResolve, closeModalReject ) => {
+				// Grab a "stable" reference of the frame element, since
+				// the value held by the react ref might change at runtime.
+				const frameEl = frameRef.current;
+
+				if ( ! frameEl ) {
+					closeModalReject();
```

Comment:
I think that rejecting is too harsh 🙂 This will most likely happen when the programmer who uses the hook makes a mistake and doesn't pass the ref to an element. Or if the element is mounted conditionally or with some delay. Then we should print a console warning ("you're using the hook wrong"), but it shouldn't prevent the modal from closing at all.

Reply:
To make sure I'm understanding your comment, do you suggest resolving here instead of rejecting?

Reply:
Yes, that's what I had in mind. If we don't have any element to animate, it's bad and probably there's some bug somewhere, but it's not a reason to abort closing the modal.